Circle CIで使用するRust nightlyバージョンをアップグレードする

2年半ぶりにCircle CIジョブを実行したところ、`mdbook test`がfailした。

- https://app.circleci.com/pipelines/github/rust-lang-ja/edition-guide/1/workflows/52a29fde-130a-4f6b-9239-14c1f16d4d9e/jobs/19

```
2021-01-10 10:18:27 [ERROR] (mdbook::utils): Error: Rustdoc returned an error: 
running 2 tests
test /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md - Raw_identifiers (line 12) ... ignored
test /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md - Raw_identifiers (line 30) ... FAILED

failures:

---- /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md - Raw_identifiers (line 30) stdout ----
error[E0658]: raw identifiers are experimental and subject to change (see issue #48589)
 --> /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md:31:4
  |
2 | fn r#match(needle: &str, haystack: &str) -> bool {
  |    ^^^^^^^
  |
  = help: add #![feature(raw_identifiers)] to the crate attributes to enable

error[E0658]: raw identifiers are experimental and subject to change (see issue #48589)
 --> /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md:36:13
  |
7 |     assert!(r#match("foo", "foobar"));
  |             ^^^^^^^
  |
  = help: add #![feature(raw_identifiers)] to the crate attributes to enable

thread '/tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md - Raw_identifiers (line 30)' panicked at 'couldn't compile the test', librustdoc/test.rs:333:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    /tmp/mdbook-u9bZ2H/rust-2018/module-system/raw-identifiers.md - Raw_identifiers (line 30)

test result: FAILED. 0 passed; 1 failed; 1 ignored; 0 measured; 0 filtered out
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: quay.io/rust-lang-ja/circleci:edition-guide
+      - image: quay.io/rust-lang-ja/circleci:edition-guide-mdbook-0.4.5
     steps:
       - checkout
       - run: rustc --version --verbose
@@ -27,6 +27,12 @@ jobs:
       - run:
           name: Testing book
           command: mdbook test
+      - run:
+          name: Check for broken links
+          command: |
+            curl -sSLo linkcheck.sh \
+              https://raw.githubusercontent.com/rust-lang/rust/master/src/tools/linkchecker/linkcheck.sh
+            sh linkcheck.sh --all edition-guide
       - run:
           name: If master branch, publish to GitHub Page
           command: |

